### PR TITLE
new Wasm spec

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
WIP. Need to bring back a few spec tests that went away in the merge.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by Cellule** [comment]
This PR seems to include the change from #1998 which is ready to merge. Why wasn't it merged before ?

File: test/WasmSpec/testsuite/imports.wast [hunk 1]
```diff
@@ -177,6 +177,7 @@
   (import "spectest" "global" (global $x i32))
   (global $y (import "spectest" "global") i32)
 
+  (import "spectest" "global" (global i64))
```
Comment (Cellule) [inline test/WasmSpec/testsuite/imports.wast]: You should remove that line.
Imported Int64 global is not legal in javascript, thus this cannot pass with our host

**Comment by Cellule** [comment]
LGTM after minor fix

File: lib/Runtime/Math/AsmJsMath.inl [hunk 2]
```diff
@@ -71,6 +71,20 @@ namespace Js
         return maxCheckNan(aLeft, aRight);
     }
 
+    template<>
+    inline double AsmJsMath::Abs<double>(double aLeft)
```
Comment (Cellule) [inline lib/Runtime/Math/AsmJsMath.inl]: Do you mean to change `aLeft` to something like `value` ?

Comment (ThomsonTan) [inline lib/Runtime/Math/AsmJsMath.inl]: That's right.

---
In reply to: [89415359](https://github.com/Microsoft/ChakraCore/pull/2050#discussion_r89415359) [](ancestors = 89415359)

**Comment by Cellule** [comment]
I think there was a problem with your rebase.
Additionally, you should reopen this PR to merge in release/1.4

**Pull request merged**